src/mesh/coordinates.cxx

```diff
@@ -1695,11 +1715,20 @@ Coordinates::FieldMetric Coordinates::Grad2_par2(const Field2D& f, CELL_LOC outl
   TRACE("Coordinates::Grad2_par2( Field2D )");
   ASSERT1(location == outloc || (outloc == CELL_DEFAULT && location == f.getLocation()));
 
-  auto invSg = 1.0 / sqrt(g_22);
-  // Communicate to get parallel slices
-  localmesh->communicate(invSg);
-  auto result = DDY(invSg, outloc, method) * DDY(f, outloc, method) * invSg
-                + D2DY2(f, outloc, method) / g_22;
+  if (invSg == nullptr) {
+    invSg = std::make_unique<FieldMetric>();
+    (*invSg) = 1.0 / sqrt(g_22);
+  }
+  if (Grad2_par2_DDY_invSg == nullptr) {
+    // Communicate to get parallel slices
+    localmesh->communicate(*invSg);
+    invSg->applyParallelBoundary("parallel_neumann");
+    // cache
+    Grad2_par2_DDY_invSg = std::make_unique<FieldMetric>();
+    (*Grad2_par2_DDY_invSg) = DDY(*invSg, outloc, method) * (*invSg);
+  }
```

Comment:
Probably worth pulling these two bits out into separate helper functions, maybe something like:

```
const FieldMetric& invSg() {
  if (invSg == nullptr) {
    invSg = std::make_unique<FieldMetric>();
    (*invSg) = 1.0 / sqrt(g_22);
  }
  return *invSg;
}
```
and then it can be used immediately as `DDY(invSg()) * invSg()`